sql/src/main/java/io/crate/execution/engine/collect/GroupByOptimizedIterator.java

```diff
@@ -236,7 +237,7 @@ static boolean hasHighCardinalityRatio(Supplier<Engine.Searcher> acquireSearcher
 
                 @Override
                 public Row apply(Map.Entry<BytesRef, Object[]> entry) {
-                    cells[0] = entry.getKey();
+                    cells[0] = BytesRefs.toString(entry.getKey());
```

Comment:
I'm uncertain, why we still have BytesRef here.

Reply:
We keep using BytesRef in the map in order to do cheap exists checks - so we don't have to create a String for each document to check if we've seen it already.